apiserver/controllers/controllers.go

```diff
@@ -82,6 +88,32 @@ func handleError(w http.ResponseWriter, err error) {
 	json.NewEncoder(w).Encode(apiErr)
 }
 
+// GetControllerInfo returns means to identify this very garm instance.
+// This is very useful for debugging and monitoring purposes.
+func (a *APIController) GetControllerInfo() (hostname, controllerId string) {
+
+	// the hostname is neither fixed nor in our control
+	// so we get it every time to avoid confusion
+	var err error
+	hostname, err = os.Hostname()
+	if err != nil {
+		log.Printf("error getting hostname: %q", err)
+		return "", ""
+	}
+
+	if a.id == "" {
```

Comment:
I think you can do this in ```NewAPIController()```, and skip this check here. You can error our in ```NewAPIController()``` if you can't get the ID. There is a call to ```maybeInitController()``` in ```main.go``` which ensures that by the time we call ```NewAPIController()``` there is an ID generated.